OpenTrons/opentrons-api
# chore: update vite version 5.3.2

# Overview
update vite version v5.3.2 and shx to v0.3.4
close AUTH-575

PD is using a few packages that have issues with `mjs`. This will be fixed in the future.

## ToDo
### update plugins
- [x] update vite.config.ts
### check make commands
- [x] make -C app dev
- [x] make -C app dev-odd
- [x] make -C app-shell-odd push-ot3 host=
- [x] make -C components dev
- [x] make -C protocol-designer dev
- [x] make -C labware-library dev
- [x] make -C labware-designer dev
- [x] make -C opentrons-ai-client dev
- [x] make test-js
- [x] make check-js
- [x] make lint-js
- [x] make -C app-shell-odd

### cjs
https://vitejs.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated
vite.config.ts -> vite.config.mts
https://opentrons.slack.com/archives/C6PF2M2LC/p1721746824847409

For vitest, the required updates will be done another PR, https://github.com/Opentrons/opentrons/pull/15687

### bonus
we can use a VSCode plugin to run test code from VSCode



# Test Plan
1. run make teardown-js && make setup-js
2. check that the above commands work 

when you run a project, there is no waring message except for `protocol-designer`

![Screenshot 2024-09-27 at 11 52 45 AM](https://github.com/user-attachments/assets/fda25904-6bc1-40fe-a188-7c5cdf237f20)


![Screenshot 2024-09-27 at 11 52 56 AM](https://github.com/user-attachments/assets/d6291bab-9fbc-46ed-b76f-c3d4a7b0c6ba)




# Changelog



# Review requests



# Risk assessment
should be low
